### ISSUE TITLE
torch

### ISSUE THREAD

**Issue by Ludobico** [Ludobico/Python_AL_01#28]
# torch.floatTensor
`torch.floatTensor`는 pytorch 라이브러리에서 정수 대신 실수를 저장하는 데이터 타입입니다. 이는 머신러닝 알고리즘에서 자주 사용되는 데이터 타입이기도 합니다.

`torch.floatTensor`는 pytorch의 기본 텐서(tensor) 타입 중 하나로, 다른 텐서 타입과 마찬가지로 수학 연산을 지원합니다.

다음은 `torch.floatTensor`의 예제 코드입니다.
```py
import torch

# 2x3 크기의 텐서 생성
tensor = torch.floatTensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])
print(tensor)

# 텐서 타입 확인
print(tensor.dtype)

# 텐서 차원 확인
print(tensor.dim())

# 텐서 크기 확인
print(tensor.size())

# 텐서의 원소 접근
print(tensor[1, 2])

# 텐서의 수학 연산
tensor2 = torch.floatTensor([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0]])
print(tensor + tensor2)
```
실행 결과는 다음과 같습니다.

```py
tensor([[1., 2., 3.],
        [4., 5., 6.]])
torch.float32
2
torch.Size([2, 3])
tensor(6.)
tensor([[2., 3., 4.],
        [5., 6., 7.]])
```

# torch.zeros
Pytorch의 torch.zeros()는 주어진 크기를 가지고 0으로 채워진 Tensor를 생성하는 함수입니다. 생성된 Tensor의 자료형은 `torch.float32`를 기본으로 사용합니다. 만약, 자료형을 지정하고 싶은 경우 dtype 인자를 사용할 수 있습니다.

다음은 torch.zeros() 함수를 사용한 예입니다.

```py
import torch

# 크기가 (3, 2)이고 자료형이 torch.float32인 Tensor를 생성합니다.
x = torch.zeros(3, 2)
print(x)  # tensor([[0., 0.],
        #          [0., 0.],
        #          [0., 0.]])

# 크기가 (3, 2)이고 자료형이 torch.int64인 Tensor를 생성합니다.
x = torch.zeros(3, 2, dtype=torch.int64)
print(x)  # tensor([[0, 0],
        #          [0, 0],
        #          [0, 0]], dtype=torch.int64)
```

# torch.mean
`torch.mean`은 pytorch에서 텐서의 평균값을 구하는 함수입니다. 텐서의 평균값은 텐서의 모든 원소의 합을 원소의 개수로 나눈 값입니다.

`torch.mean` 함수는 다음과 같이 사용할 수 있습니다.
```py
import torch

# 텐서 생성
tensor = torch.floatTensor([1.0, 2.0, 3.0, 4.0])

# 텐서의 평균값 구하기
mean = torch.mean(tensor)
print(mean)  # tensor(2.5000)
```

`torch.mean` 함수는 축(axis) 인자를 지정할 수 있습니다. 이 인자를 지정하면 텐서의 해당축에 대해 평균값을 구할 수 있습니다. 예를 들어, 아래 코드는 행렬의 각 열의 평균값을 구합니다.

```py
import torch

# 2x3 행렬 생성
matrix = torch.floatTensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])

# 각 열의 평균값 구하기
mean = torch.mean(matrix, dim=0)
print(mean)  # tensor([2.5000, 3.5000, 4.5000])
```

# torch.set_grad_enabled
`torch.set_grad_enabled` 함수는 `Pytorch`의 자동 미분(autograd) 기능을 켜거나 끌 수 있는 함수입니다. 이 기능을 켜면 기본적으로 Pytorch의 연산은 자동으로 역전파(Backpropagation)를 수행합니다.

`torch.set_grad_enabled` 함수는 아래와 같이 사용합니다.
```py
torch.set_grad_enabled(mode)
```

`mode`는 `True` 또는 `False` 값을 가집니다. `True`이면 자동 미분 기능을 켜고, `False`면 자동 미분 기능을 끕니다.

예를 들어, 아래 코드는 자동 미분 기능을 켜고 계산을 수행한 후 자동 미분 기능을 끕니다.

```py
import torch

# 자동 미분 기능을 켬
torch.set_grad_enabled(True)

x = torch.tensor(1.0, requires_grad=True)
y = x ** 2
y.backward()
print(x.grad)  # 2.0

# 자동 미분 기능을 끔
torch.set_grad_enabled(False)
```

# .item()
`item()`은 텐서에서 스칼라 값을 추출하기 위해 사용합니다. 텐서가 스칼라 값일 경우 item() 메서드를 호출하면 그 값을 추출할 수 있습니다.

다음은 item() 메서드의 예제 코드입니다.

```py
import torch

# 텐서를 정의합니다.
tensor = torch.tensor(1.5)

# 텐서에서 스칼라 값을 추출합니다.
scalar_value = tensor.item()
print(scalar_value)  # 1.5

# 텐서의 차원이 1일 경우 item()을 사용할 수 없습니다.
tensor = torch.tensor([1.5])
try:
    scalar_value = tensor.item()
except:
    print("tensor.item() is not defined for tensors with more than one element.")
```

item() 메서드는 텐서가 스칼라 값일 경우에만 사용할 수 있습니다. 텐서의 차원이 1일 경우 item() 메서드를 사용하면 에러가 발생합니다.

만약 텐서의 차원이 1이고, 첫번째 값만 추출하고 싶다면, `tensor[0]`를 사용할 수 있습니다.

# torch.Tensor
`torch.Tensor`는 PyTorch에서 기본적으로 제공하는 고정 길이의 다차원 배열 자료형입니다. 이것은 기본적으로 실수 값을 가지고 있지만, 정수 값을 가질 수도 있습니다.

`torch.Tensor`는 여러 개의 파라미터로 생성할 수 있습니다. 예를 들어, 다음과 같은 파라미터를 지정할 수 있습니다.

* `data` : 생성할 `torch.Tensor`에 저장할 데이터
* `dtype` : 데이터 타입
* `device` : 저장될 위치 (CPU 또는 GPU)
* `requires_grad` : 기울기를 가지는지 여부

예를 들어, 다음과 같이 `torch.Tensor`를 생성할 수 있습니다.
```py
import torch

# 1x2x3 크기의 텐서 생성
data = [[[1, 2, 3], [4, 5, 6]]]
tensor = torch.Tensor(data)

# 텐서의 크기 확인
print(tensor.size())  # prints torch.Size([1, 2, 3])

# GPU 상에서 텐서 생성
device = torch.device('cuda')
tensor = torch.Tensor(data, device=device)

# 기울기를 가지는 텐서 생성
tensor = torch.Tensor(data, requires_grad=True)
```
